components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/HTTPAnalyticsResponseCodeFilter.jsx

```diff
@@ -236,25 +309,73 @@ class HTTPAnalyticsResponseCodeFilter extends Widget {
         data.data.forEach((dataUnit) => {
             services.push(dataUnit[1]);
         });
-
         services = services.filter((v, i, a) => a.indexOf(v) === i);
+        services.sort(this.naturalSort);
+
         const serviceOptions = services.map(service => ({
             value: service,
             label: service,
             disabled: false,
         }));
 
         this.setState({
-            services, serviceOptions,
-        }, this.publishUpdate);
+            services,
+            serviceOptions,
+        }, this.initialPublish);
+    }
+
+    /**
+     * Handle initial selection publish
+     */
+    initialPublish() {
+        const responseCodeSelection = super.getGlobalState('httpResCode');
+        if (responseCodeSelection.responseCode
+            && !(responseCodeSelection.responseCode instanceof Array)
+            && this.state.services.indexOf(responseCodeSelection.responseCode) !== -1) {
+            this.handleChange({
+                value: responseCodeSelection.responseCode,
+                label: responseCodeSelection.responseCode,
+                disabled: false,
+            });
+        } else {
+            this.publishUpdate();
+        }
+    }
+
+    /**
+     * Sort in ascending order
+     * @param a
+     * @param b
+     */
+    naturalSort(a, b) {
```

Comment:
Change as suggested in code review

Reply:
This is the comparator function passed to the sort method, not a sort function.